### ISSUE TITLE
cannot find namespace firebase

### ISSUE THREAD

**Issue by alizzen** [angular/angularfire2#529]
<img width="1057" alt="screen shot 2016-09-18 at 11 07 21 pm" src="https://cloud.githubusercontent.com/assets/16546209/18618710/caaaf8aa-7df4-11e6-9522-0b0b9e34eccd.png">
<img width="1680" alt="screen shot 2016-09-18 at 11 08 37 pm" src="https://cloud.githubusercontent.com/assets/16546209/18618715/d7c0fa26-7df4-11e6-857e-06380a806914.png">

please let me know if using the rest api of firebase is better

**Comment by mukesh51** [comment]
add the following entry in src/tsconfig.json to get rid of " Cannot find namespace 'firebase" error

"types": [
      "firebase"
    ]

Look at the troubleshooting section of tutorial [installation](https://github.com/angular/angularfire2/blob/master/docs/1-install-and-setup.md),  for more information around other errors, you may encounter.

**Comment by thebishorup** [comment]
It still does not solve the problem for me in Angular 2 (RC7).

**Comment by mukesh51** [comment]
Why not move away from RC7 to a more stable version of angular2. It seems more stable. Rather follow the install tutorial again from start or update your package.json to something like this

dependencies": {
    "@angular/common": "2.0.0",
    "@angular/compiler": "2.0.0",
    "@angular/core": "2.0.0",
    "@angular/forms": "2.0.0",
    "@angular/http": "2.0.0",
    "@angular/platform-browser": "2.0.0",
    "@angular/platform-browser-dynamic": "2.0.0",
    "@angular/router": "3.0.0",
    "angularfire2": "^2.0.0-beta.4",
    "core-js": "^2.4.1",
    "firebase": "^3.4.0",
    "rxjs": "5.0.0-beta.12",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.6.23"
  },

Hopefully this should get you going.

**Comment by tonymk** [comment]
Same problem with this setup:
Windows 10, angular-cli beta 14
{
  "dependencies": {
    "@angular/common": "2.0.0",
    "@angular/compiler": "2.0.0",
    "@angular/core": "2.0.0",
    "@angular/forms": "2.0.0",
    "@angular/http": "2.0.0",
    "@angular/platform-browser": "2.0.0",
    "@angular/platform-browser-dynamic": "2.0.0",
    "@angular/router": "3.0.0",
    "@types/request": "0.0.30",
    "angularfire2": "^2.0.0-beta.5",
    "core-js": "^2.4.1",
    "firebase": "^3.4.0",
    "rxjs": "5.0.0-beta.12",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.6.23"
  },
  "devDependencies": {
    "@types/jasmine": "^2.2.30",
    "angular-cli": "1.0.0-beta.14",
    "codelyzer": "~0.0.26",
    "jasmine-core": "2.4.1",
    "jasmine-spec-reporter": "2.5.0",
    "karma": "1.2.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.0.2",
    "karma-remap-istanbul": "^0.2.1",
    "protractor": "4.0.5",
    "ts-node": "1.2.1",
    "tslint": "3.13.0",
    "typescript": "2.0.2"
  }
}
And i have added: 
"types": [
"firebase"
]

**Comment by mukesh51** [comment]
@tonymk Just curious. Did you update the package.json file and ran npm install or did you do a full install from scratch. May be worth trying a later. I've to admit, I had my own issues on Windows 7 and Windows 12 server, but everything worked smoothly on Ubuntu 14.04. I am using EC2 instances.

Hopefully @davideast  should be able to help here.

**Comment by tonymk** [comment]
Used angular-cli to make a new project, then a npm install -g firebase angularfire2.
Then added angularfire2 module to the app.module

**Comment by samwah** [comment]
Hi, I'm running Angular 2.0.0 with angular-cli 1.0.0-beta.14, this worked for me `¯\_(ツ)_/¯`:
Add to "package.json":
```
"dependencies": {
...
    "@types/firebase": "^2.4.29",
    "@types/request": "0.0.30",
...
}
```
Add to "tsconfig.json":
```
"compilerOptions": {
...
    "types": [
    "firebase"
]
}
```
Add to "typings.d.ts":
```
...
declare var require: any;
declare var module: any;
````

**Comment by dmastag** [comment]
I am seeing a trend where adding types is not solving the issue on Windows 10.
@mukesh51 are you on Win 10 by any chance ?

**Comment by Soolan** [comment]
Hi guys,
adding types: ['firebase'] solved my problem on my ubuntu 16 machine but now I get the following error:
`Uncaught SyntaxError: Unexpected token export`
I'm using a fesh https://github.com/angular/angular2-seed.git so everything is the latest. 

I also noticed the following warning:
`UNMET PEER DEPENDENCY @types/request@0.0.30`

Any idea what might be wrong?
Thank you all

**Comment by katowulf** [comment]
@soolan Let's keep this discussion on topic. Submit your separate errors as a issue. Thanks!

**Comment by katowulf** [comment]
Thanks to everyone for noting these errors. If you're experiencing this error after adding `types: ['firebase']` then please do the following:

 - double-check your angular-cli output for any errors (and share them)
 - note all of the following in your comment:  `ng -v`, `node -v`, `npm list --depth=0`, `npm list -g --depth=0`, your environment info, and the complete contents of your `package.json` and `src/tsconfig.json` files.
 - if any of those things are not on the latest supported version according to our install guide, stop and upgrade before commenting here.

**Comment by ianrothmann** [comment]
I solved my issue by adding:
/// <reference path="../node_modules/firebase/firebase.d.ts" />
to typings.d.ts

**Comment by thebishorup** [comment]
Adding the following on typings.d.ts solved the problem for me as well.

/// <reference path="../node_modules/firebase/firebase.d.ts" />

**Comment by anderflash** [comment]
Another solution is including in app.module.ts:
~~~typescript
import * as firebase from 'firebase';
~~~

**Comment by jeffbcross** [comment]
This should be fixed by c3a954cd857fc297438f53d3a1e3bf69b6af01c2
